### ISSUE TITLE
Bug in Helm version 1.13.0 & 1.13.1

### ISSUE THREAD

**Issue by m-over** [argoproj/argo-cd#16320]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**
In versions, 1.13.0 & 1.13.1 is a bug that, causes an issue that it does not pull a Helm Chart from an Azure ACR. The bug is solved in the version 1.13.2. 

One of the issues in the Helm project: [https://github.com/helm/helm/issues/12491](https://github.com/helm/helm/issues/12491)
The reverted commit in version: [https://github.com/helm/helm/commit/e785e6c50c622ed5019fd4020a13509c87a3022d](https://github.com/helm/helm/commit/e785e6c50c622ed5019fd4020a13509c87a3022d)

**To Reproduce**
1) Create a ACR
2) Create Helm chart and push to ACR
3) Create ACR access token
4) Add ACR repo to argocd with OCI enabled
5) Try to create a app in argo -> Error Message with 404 Unauthorized

**Expected behavior**
Helm an pull the chart from the repo

**Version**
```shell
argocd: v2.9.0+9cf0c69.dirty
  BuildDate: 2023-11-06T15:07:31Z
  GitCommit: 9cf0c69bbe70393db40e5755e34715f30179ee09
  GitTreeState: dirty
  GoVersion: go1.21.3
  Compiler: gc
  Platform: darwin/arm64
argocd-server: v2.9.0+edc213d
  BuildDate: 2023-11-11T18:48:14Z
  GitCommit: edc213d22902e9676231119522563fa338c62793
  GitTreeState: clean
  GoVersion: go1.21.3
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v5.2.1 2023-10-19T20:13:51Z
  Helm Version: v3.13.1+g3547a4b
  Kubectl Version: v0.26.4
  Jsonnet Version: v0.20.0
```

**Comment by visgde** [comment]
We're experiencing the same problem here => upvote

**Comment by blakepettersson** [comment]
Fixed with #16321
